### PR TITLE
Compile patterns once

### PR DESCRIPTION
**Pull request by carlmontanari**
follow up from #56 

wrapped all the singleton-y pattern compiling pieces with sync.Once and added a test to make sure that the channel auth patterns fetcher wont cause race condition. had to export some stuff for the patterns business to test but seems like thats not the worst thing ever.

@hellt @networkop does this look reasonable to you gents?

### CODE REVIEW

File: cfg/aristaeos.go [hunk 1]
```diff
@@ -18,16 +19,19 @@ type eosPatterns struct {
 	endPattern               *regexp.Regexp
 }
 
-var eosPatternsInstance *eosPatterns //nolint:gochecknoglobals
+var (
+	eosPatternsInstance     *eosPatterns //nolint:gochecknoglobals
+	eosPAtternsInstanceOnce sync.Once    //nolint:gochecknoglobals
```
Comment (hellt) [inline cfg/aristaeos.go:24]: typoe `PA->Pa`

**Comment by hellt** [comment]
lgtm

**Review by networkop — APPROVED** [review]
lgtm

File: cfg/aristaeos.go (hunk 1 above)
Comment (carlmontanari) [reply cfg/aristaeos.go]: good catch, thanks dude!

**Comment by carlmontanari** [comment]
Awesome! Thank you both very much! Thanks @networkop for raising this and the help!

Carl

**Pull request merged**